cdc/sink/mysql.go

```diff
@@ -88,17 +76,23 @@ func configureSinkURI(sinkURI string) (string, error) {
 	return dsnCfg.FormatDSN(), nil
 }
 
-// NewMySQLSinkUsingSchema creates a new MySQL sink
-func NewMySQLSinkUsingSchema(db *sql.DB, schemaStorage *schema.Storage) Sink {
-	inspector := &cachedInspector{
-		db:    db,
-		cache: make(map[string]*tableInfo),
-		tableGetter: func(_ *sql.DB, schemaName string, tableName string) (*tableInfo, error) {
-			info, err := getTableInfoFromSchemaStorage(schemaStorage, schemaName, tableName)
+// NewMySQLSinkUsingSchema creates a new MySQL sink using schema storage
+func NewMySQLSinkUsingSchema(sinkURI string, infoGetter TableInfoGetter, _ map[string]string) (Sink, error) {
+	sinkURI, err := configureSinkURI(sinkURI)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	db, err := sql.Open("mysql", sinkURI)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	inspector := &tableInspectorImpl{
```

Comment:
We can add methods like `GetUniqueKeys` to `schema.TableInfo` and get rid of the `tableInspector` interface completely.

Reply:
good idea, I will implement this in next PR

Reply:
OK